tuning/config/acceleration_configs/acceleration_framework_config.py

```diff
@@ -91,13 +92,48 @@ class AccelerationFrameworkConfig:
     fast_kernels: Annotated[
         FastKernelsConfig,
         ConfigAnnotation(
-            path="peft.quantization",
+            path="training",
             key="fused_ops_and_kernels",
             experimental=True,
             required_packages=["foak"],
         ),
     ] = None
 
+    padding_free: Annotated[
+        PaddingFree,
+        ConfigAnnotation(
+            path="training.attention",
+            experimental=True,
+            required_packages=["aadp"],
+        ),
+    ] = None
+
+    multipack: Annotated[
+        MultiPack,
+        ConfigAnnotation(
+            path="training.dataloader",
+            experimental=True,
+            required_packages=["aadp"],
+        ),
+    ] = None
+
+    def _verify_configured_dataclasses(self):
+        if self.multipack is not None:
+            # ensure if multipack is set, padding free is also turned on as well
+            # this also ensures that the attention implementation for multipack
+            # will be flash attention as sfttrainer will enforce flash attn to be
+            # set for padding free
+            assert (
+                self.padding_free is not None
+            ), "`--multipack` is currently only supported with `--padding_free`"
+
+        # Check that fused lora must be activated with either auto_gptq or bitsandbytes
+        if self.fused_lora is not None:
+            assert self.bitsandbytes is not None or self.auto_gptq is not None, (
+                "`--fused_lora` must be accompanied by a quantized base layer"
+                " `--auto_gptq` or `--bitsandbytes`."
```

Comment:
I appreciate the checks but think these should be raised ValueErrors instead of AssertionErrors

Reply:
ok @achew010 can you help make the changes?